xdsl/irdl/attributes.py

```diff
@@ -256,8 +262,30 @@ def irdl_attr_definition(cls: TypeAttributeInvT) -> TypeAttributeInvT:
     )
 
 
+IRDLGenericAttributeConstraint: TypeAlias = (
+    GenericAttrConstraint[AttributeInvT]
+    | Attribute
+    | type[AttributeInvT]
+    | "TypeForm[AttributeInvT]"
+    | ConstraintVar
+    | TypeVar
+)
+"""
+Attribute constraints represented using the IRDL python frontend. Attribute constraints
+can either be:
+- An instance of `AttrConstraint` representing a constraint on an attribute.
```

Comment:
Would it be worth documenting the effect of an attribute constraint in a variadic? i.e. using `var_operand_def(my_attr_constr)` gives you a `RangeOf` constraint where each element of the range is constrained by `my_attr_constr`

Reply:
Why would this be documented here? I feel this kind of documentation should be on the `var_operand_def` and al rather than there, or on attribute constraints directly?

Reply:
Looking at it again that seems fair